Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -3472,7 +3473,11 @@ else if (thisReq_ instanceof ServerRequestLogout) {
                 } catch (JSONException ex) {
                     ex.printStackTrace();
                 }
+                if (thisReq_ instanceof ServerRequestInitSession) {
+                    branchUniversalReferralInitWrapper_ = null;
```

Comment:
The call back is nullified only after response is sent for "initSession". Since there is an   additional async operation  `fetchDeferredAppLinkData` the callback wrapper need a strong reference until  the request queued